pkg/kdk/restart.go

```diff
@@ -0,0 +1,64 @@
+// Copyright © 2018 Cisco Systems, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kdk
+
+import (
+	"fmt"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/cisco-sso/kdk/pkg/prompt"
+	"github.com/docker/docker/api/types"
+)
+
+func Restart(cfg KdkEnvConfig, logger logrus.Entry) error {
+
+	var containerIds []string
+
+	containers, err := cfg.DockerClient.ContainerList(cfg.Ctx, types.ContainerListOptions{})
+
+	if err != nil {
+		logger.WithField("error", err).Fatal("Failed to list docker containers")
+	}
+	for _, container := range containers {
+		for _, name := range container.Names {
+			if name == "/"+cfg.ConfigFile.AppConfig.Name {
+				containerIds = append(containerIds, container.ID)
+				break
+			}
+		}
+	}
+	if len(containerIds) > 0 {
+		logger.Info("Restarting KDK container(s)...")
+		for _, containerId := range containerIds {
+			fmt.Printf("Restart KDK container [%s][%v]\n", cfg.ConfigFile.AppConfig.Name, containerId[:8])
+			prmpt := prompt.Prompt{
+				Text:     "Continue? [y/n] ",
+				Loop:     true,
+				Validate: prompt.ValidateYorN,
+			}
+			if result, err := prmpt.Run(); err != nil || result == "n" {
+				logger.Error("KDK container restart canceled or invalid input.")
+				return nil
+			}
+			if err := cfg.DockerClient.ContainerRestart(cfg.Ctx, containerId); err != nil {
```

Comment:
I think in the old Python kdk version the restart might have been a snapshot and then run again, which saved state.   I don't think we're doing this here, and I'm not sure if it is what people expect when "restarting".